daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/FNFunctions.scala

```diff
@@ -803,6 +805,85 @@ case class FNFloor(recipe: CompiledDPath, argType: NodeInfo.Kind) extends FNOneA
   }
 }
 
+/**
+ * This is "arithmetic" shift left, meaning it sign-extends a negative signed value.
```

Comment:
Hmm. There is no such thing as arithmetic shift left. Only right shifts have the distinction of arithmetic (sign extending) and logical (no sigh extend). 

Update comment.